Source/WebKit/Shared/Cocoa/CoreIPCNSURLCredential.mm

```diff
@@ -26,9 +26,249 @@
+                ASSERT_NOT_REACHED();
+                break;
+            }
+            NSString *value = [flags objectForKey:key];
+            if (![value isKindOfClass:NSString.class]) {
+                ASSERT_NOT_REACHED();
+                break;
+            }
+            auto k = CoreIPCString(key);
+            auto v = CoreIPCString(value);
+            vector.append(std::make_pair(WTFMove(k), WTFMove(v)));
+        }
+        m_data.flags = WTFMove(vector);
+    }
+
+    SET_MDATA(uuid, NSString, CoreIPCString);
+    SET_MDATA(appleID, NSString, CoreIPCString);
+    SET_MDATA(realm, NSString, CoreIPCString);
+    SET_MDATA(token, NSString, CoreIPCString);
+
+    NSNumber *type = dict[@"type"];
+    if ([type isKindOfClass:NSNumber.class]) {
+        switch ([type intValue]) {
+        case kURLCredentialInternetPassword:
+            m_data.type = CoreIPCNSURLCredentialType::Password;
+            break;
+        case kURLCredentialServerTrust:
+            m_data.type = CoreIPCNSURLCredentialType::ServerTrust;
+            break;
+        case kURLCredentialKerberosTicket:
+            m_data.type = CoreIPCNSURLCredentialType::KerberosTicket;
+            break;
+        case kURLCredentialXMobileMeAuthToken:
+            m_data.type = CoreIPCNSURLCredentialType::XMobileMeAuthToken;
+            break;
+        case kURLCredentialOAuth2:
+            m_data.type = CoreIPCNSURLCredentialType::OAuth2;
+            break;
+        default:
+            ASSERT_NOT_REACHED();
+            m_data.type = CoreIPCNSURLCredentialType::Password;
+            break;
+        }
+    }
+}
+
+CoreIPCNSURLCredential::CoreIPCNSURLCredential(CoreIPCNSURLCredentialData&& data)
+    : m_data(WTFMove(data)) { }
+
+RetainPtr<id> CoreIPCNSURLCredential::toID() const
+{
+    auto dict = adoptNS([[NSMutableDictionary alloc] initWithCapacity:7]);
+
+    RetainPtr<NSNumber> persistence;
+    switch (m_data.persistence) {
+    case CoreIPCNSURLCredentialPersistence::None:
+        persistence = adoptNS([NSNumber numberWithInt:kCFURLCredentialPersistenceNone]);
+        break;
+    case CoreIPCNSURLCredentialPersistence::Session:
+        persistence = adoptNS([NSNumber numberWithInt:kCFURLCredentialPersistenceForSession]);
+        break;
+    case CoreIPCNSURLCredentialPersistence::Permanent:
+        persistence = adoptNS([NSNumber numberWithInt:kCFURLCredentialPersistencePermanent]);
+        break;
+    case CoreIPCNSURLCredentialPersistence::Synchronizable:
+        persistence = adoptNS([NSNumber numberWithInt:kCFURLCredentialPersistenceSynchronizable]);
+        break;
+    default:
+        ASSERT_NOT_REACHED();
+        persistence = adoptNS([NSNumber numberWithInt:kCFURLCredentialPersistenceNone]);
+        break;
+    }
+    [dict setObject:persistence.get() forKey:@"persistence"];
+
+    switch (m_data.type) {
+    case CoreIPCNSURLCredentialType::Password:
+        if (m_data.user && m_data.password) {
```

Comment:
We should not need to check the existence of members now that type is not inferred. If we want to be sure user and password exist, we could add assertion for them. Like:
```
[dict setObject:@(kURLCredentialInternetPassword) forKey:@"type"];
ASSERT(m_data.user);
if (m_data.user)
    [dict setObject:(*m_data.user).toID().get() forKey:@"user"];
ASSERT(m_data.password);
if (m_data.password)
    [dict setObject:(*m_data.password).toID().get() forKey:@"password"];
...
if (m_data.useKeychain)
    [dict setObject:[NSNumber numberWithBool:(*m_data.useKeychain)] forKey:@"useKeychain"];
break;
```